openedx/core/djangoapps/zendesk_proxy/tests/test_v1_views.py

```diff
@@ -0,0 +1,128 @@
+
+from openedx.core.lib.api.test_utils import ApiTestCase
+from openedx.core.djangoapps.zendesk_proxy.v0.views import ZENDESK_REQUESTS_PER_HOUR
+
+
+@ddt.ddt
+@override_settings(
+    ZENDESK_URL="https://www.superrealurlsthataredefinitelynotfake.com",
+    ZENDESK_OAUTH_ACCESS_TOKEN="abcdefghijklmnopqrstuvwxyz1234567890"
+)
+class ZendeskProxyTestCase(ApiTestCase):
+    """Tests for zendesk_proxy views."""
+
+    def setUp(self):
+        self.url = reverse('zendesk_proxy_v0')
+        self.request_data = {
+            'requester': {
+                'email': 'JohnQStudent@example.com',
+                'name': 'John Q. Student'
+            },
+            'subject': 'Python Unit Test Help Request',
+            'comment': {
+                'body': "Help! I'm trapped in a unit test factory and I can't get out!",
+            },
+            'tags': ['python_unit_test'],
+            'custom_fields': [
+                {
+                    'id': '001',
+                    'value': 'demo-course'
+                }
+            ],
+        }
+        return super(ZendeskProxyTestCase, self).setUp()
+
+    def test_post(self):
+        with patch('requests.post', return_value=MagicMock(status_code=201)) as mock_post:
+            response = self.request_without_auth(
+                'post',
+                self.url,
+                data=json.dumps(self.request_data),
+                content_type='application/json'
+            )
+            self.assertHttpCreated(response)
+            (mock_args, mock_kwargs) = mock_post.call_args
+            self.assertEqual(mock_args, ('https://www.superrealurlsthataredefinitelynotfake.com/api/v2/tickets.json',))
+            self.assertEqual(
+                mock_kwargs,
+                {
+                    'headers': {
+                        'content-type': 'application/json',
+                        'Authorization': 'Bearer abcdefghijklmnopqrstuvwxyz1234567890'
+                    },
+                    'data': '{"ticket": {"comment": {"body": "Help! I\'m trapped in a unit test factory and I can\'t get out!", "uploads": null}, "tags": ["python_unit_test"], "subject": "Python Unit Test Help Request", "custom_fields": [{"id": "001", "value": "demo-course"}], "requester": {"name": "John Q. Student", "email": "JohnQStudent@example.com"}}}'  # pylint: disable=line-too-long
+                }
+            )
+
+    @ddt.data('requester', 'tags')
+    def test_bad_request(self, key_to_delete):
+        test_data = deepcopy(self.request_data)
+        _ = test_data.pop(key_to_delete)
+
+        response = self.request_without_auth(
+            'post',
+            self.url,
+            data=json.dumps(test_data),
+            content_type='application/json'
+        )
+        self.assertHttpBadRequest(response)
+
+    @override_settings(
+        ZENDESK_URL=None,
+        ZENDESK_OAUTH_ACCESS_TOKEN=None
+    )
+    def test_missing_settings(self):
```

Comment:
This test isn't any different for v1 as opposed to v0, the 503 condition is thrown from `create_zendesk_ticket`. It can be removed here.

Reply:
This actually applies to `test_zendesk_status_codes` and `test_unexpected_error_pinging_zendesk` as well.

Would you mind doing a little refactoring here? These three tests should really be separated into a new file, `test_utils.py` or similar. Then they can call `create_zendesk_ticket` directly (as opposed to `self.request_without_auth`) and perform these checks.

This will keep view-specific tests in `test_v1_views.py` and `test_v0_views.py`